faiss_vector_posting.go

```diff
@@ -403,3 +403,25 @@ func (sb *SegmentBase) InterpretVectorIndex(field string) (segment.VectorIndex,
 	vecIndex, err = faiss.ReadIndexFromBuffer(indexBytes, faiss.IOFlagReadOnly)
 	return wrapVecIndex, err
 }
+
+func (sb *SegmentBase) UpdateFieldStats(stats map[string]map[string]int) {
+
+	for _, fieldName := range sb.fieldsInv {
+
+		pos := int(sb.fieldsSectionsMap[sb.fieldsMap[fieldName]-1][SectionFaissVectorIndex])
+		if pos == 0 {
+			continue
+		}
+
+		_, n := binary.Uvarint(sb.mem[pos : pos+binary.MaxVarintLen64])
+		pos += n
+		_, n = binary.Uvarint(sb.mem[pos : pos+binary.MaxVarintLen64])
+		pos += n
+		_, n = binary.Uvarint(sb.mem[pos : pos+binary.MaxVarintLen64])
+		pos += n
+
+		numVecs, _ := binary.Uvarint(sb.mem[pos : pos+binary.MaxVarintLen64])
+
+		stats["num_vectors"][fieldName] = int(numVecs)
```

Comment:
I see you're using "num_vectors" in both the zapx and bleve PRs - I think this can be declared once and re-used.

Reply:
Added stuff in scorch_segment_api to track the stat names